tests/PositionTests.cpp

```diff
@@ -35,12 +35,22 @@ TEST_CASE("Null Move Test", "[Position]") {
 
 TEST_CASE("Hash Test", "[Position]") {
     Position pos{STARTPOS_FEN};
+
     Position::hash_type old_hash = pos.hash();
 
     pos.make_move({E2, E4});
     REQUIRE(pos.hash() != old_hash);
     pos.unmake_move();
     REQUIRE(pos.hash() == old_hash);
+
+    pos.make_move({B1, C3});
+    REQUIRE(pos.hash() != old_hash);
+    pos.make_move({B8, C6});
+    REQUIRE(pos.hash() != old_hash);
+    pos.make_move({C3, B1});
+    REQUIRE(pos.hash() != old_hash);
+    pos.make_move({C6, B8});
+    REQUIRE(pos.hash() == old_hash);
```

Comment:
Add a new test case which checks whether the make_move (in each of the cases you added incremental hash) causes new hash to be equal to calculate_hash() from scratch. This proves it's equivalent to the old functionality.

Reply:
Calculate_hash is a private member. Can I make it public?

Reply:
Yeah I guess it's fine

Reply:
Note that even with this partial hash-update, the perft-batch went from 01:55 to 01:40 on my laptop.